<img src="https://1000logos.net/wp-content/uploads/2017/11/Zillow-Logo.png" title="Zillow Logo"/>

***

- imports that we will be using for this data set

In [1]:
# fetches the data
import acquire
# credentials file to access the data
import env
# Imports functions necessary to run visuals and hides unnecessary code
import wrangle

# coding 
import math
import numpy as np
import seaborn as sns
import scipy.stats as stats
import pandas as pd
import matplotlib.pyplot as plt
from pydataset import data
import scipy.stats
import scipy
import os

# needed for modeling
import sklearn.preprocessing
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier
from sklearn.preprocessing import MinMaxScaler, StandardScaler, RobustScaler, QuantileTransformer
from sklearn.metrics import explained_variance_score
import statsmodels.api as sm
from sklearn.metrics import mean_squared_error
from sklearn.linear_model import LinearRegression, LassoLars, TweedieRegressor
from sklearn.preprocessing import PolynomialFeatures
from sklearn.cluster import KMeans

***

Fips dictionary:
- 6037.0 = Los Angeles,CA
- 6059.0 = Orange,CA
- 6111.0 = Ventura,CA

# <span style="color:blue">Plan</span>
- Acquire Data
- Clean and Prep the Data
- Explore the features of the Zillow data
- Evaluate Data
- Test Data

# <span style="color:blue">Acquiring Data</span>

In [2]:
# importing and aquiring data set
df = acquire.get_zillow_data()

### - Summary

In [3]:
# summary function for DataFrame
wrangle.summarize(df)

--- Shape: (71858, 69)
--- Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71858 entries, 0 to 71857
Data columns (total 69 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   propertylandusetypeid         71858 non-null  float64
 1   parcelid                      71858 non-null  int64  
 2   storytypeid                   47 non-null     float64
 3   typeconstructiontypeid        223 non-null    float64
 4   heatingorsystemtypeid         46680 non-null  float64
 5   buildingclasstypeid           0 non-null      float64
 6   architecturalstyletypeid      207 non-null    float64
 7   airconditioningtypeid         23069 non-null  float64
 8   id                            71858 non-null  int64  
 9   basementsqft                  47 non-null     float64
 10  bathroomcnt                   71858 non-null  float64
 11  bedroomcnt                    71858 non-null  float64
 12  buildingqualitytypeid      

***

# <span style="color:blue">Preparing & Cleaning Data</span>
- all these functions will be found in the explore.py file

###  What percentage of data is missing per column?

In [4]:
#looking at percentage of null values by column
wrangle.nulls_by_columns(df).sort_values(by= 'percent', ascending=False)

count   percent
buildingclasstypeid           71858  1.000000
buildingclassdesc             71858  1.000000
finishedsquarefeet13          71856  0.999972
finishedsquarefeet15          71847  0.999847
storydesc                     71811  0.999346
basementsqft                  71811  0.999346
storytypeid                   71811  0.999346
yardbuildingsqft26            71788  0.999026
finishedsquarefeet6           71692  0.997690
fireplaceflag                 71686  0.997606
architecturalstyletypeid      71651  0.997119
architecturalstyledesc        71651  0.997119
typeconstructiontypeid        71635  0.996897
typeconstructiondesc          71635  0.996897
pooltypeid10                  71394  0.993543
decktypeid                    71269  0.991803
poolsizesum                   70989  0.987907
pooltypeid2                   70784  0.985054
hashottuborspa                70320  0.978597
yardbuildingsqft17            69608  0.968688
taxdelinquencyyear            69243  0.963609
taxdelinquencyflag            69243  0.963609
finishedsquarefeet50          66161  0.920719
finishedfloor1squarefeet      66161  0.920719
fireplacecnt                  63754  0.887222
threequarterbathnbr           61859  0.860851
pooltypeid7                   57193  0.795917
poolcnt                       56101  0.780720
numberofstories               54812  0.762782
airconditioningtypeid         48789  0.678964
airconditioningdesc           48789  0.678964
garagetotalsqft               46887  0.652495
garagecarcnt                  46887  0.652495
regionidneighborhood          43621  0.607044
buildingqualitytypeid         26795  0.372888
unitcnt                       26422  0.367697
propertyzoningdesc            26378  0.367085
heatingorsystemdesc           25178  0.350385
heatingorsystemtypeid         25178  0.350385
lotsizesquarefeet              8126  0.113084
regionidcity                   1336  0.018592
finishedsquarefeet12            335  0.004662
censustractandblock             227  0.003159
calculatedbathnbr               223  0.003103
fullbathcnt                     223  0.003103
yearbuilt                       191  0.002658
calculatedfinishedsquarefeet    156  0.002171
structuretaxvaluedollarcnt       94  0.001308
regionidzip                      44  0.000612
taxamount                         5  0.000070
landtaxvaluedollarcnt             2  0.000028
taxvaluedollarcnt                 1  0.000014
id.1                              0  0.000000
logerror                          0  0.000000
transactiondate                   0  0.000000
propertylandusetypeid             0  0.000000
assessmentyear                    0  0.000000
roomcnt                           0  0.000000
regionidcounty                    0  0.000000
rawcensustractandblock            0  0.000000
propertycountylandusecode         0  0.000000
parcelid                          0  0.000000
longitude                         0  0.000000
latitude                          0  0.000000
fips                              0  0.000000
bedroomcnt                        0  0.000000
bathroomcnt                       0  0.000000
id                                0  0.000000
propertylandusedesc               0  0.000000

### Takeaways:

- we can see that there is a large percentage of information missing in alot of the columns (ranging from column regionidneighborhood with %60 data missing to buildingclasstypeid with %100 of the data missing.

### Functions to clean data:

- lets build a function to address the columns with large percentage of missing data.

In [9]:
k = 1.5
cols = ['bathroomcnt', 'bedroomcnt','calculatedfinishedsquarefeet','yearbuilt','lotsizesquarefeet','lotsizesquarefeet']    

def handle_outliers(df, cols, k):
    """this will eliminate most outliers, use a 1.5 k value if unsure because it is the most common, make sure to define cols value as the features
    you want the outliers to be handled. this should be done before running the function and outiside of it"""

    
    # Create placeholder dictionary for each columns bounds
    bounds_dict = {}
   
    for col in cols:
        # get necessary iqr values
        q1 = df[col].quantile(0.25)
        q3 = df[col].quantile(0.75)
        iqr = q3 - q1
        upper_bound =  q3 + k * iqr
        lower_bound =  q1 - k * iqr

        #store values in a dictionary referencable by the column name
        #and specific bound
        bounds_dict[col] = {}
        bounds_dict[col]['upper_bound'] = upper_bound
        bounds_dict[col]['lower_bound'] = lower_bound

    for col in cols:
        #retrieve bounds
        col_upper_bound = bounds_dict[col]['upper_bound']
        col_lower_bound = bounds_dict[col]['lower_bound']

        #remove rows with an outlier in that column
        df = df[(df[col] < col_upper_bound) & (df[col] > col_lower_bound)]
        
    return df
df = handle_outliers(df, cols, k)
df

propertylandusetypeid  parcelid  storytypeid  typeconstructiontypeid  \
0                      261.0  14297519          NaN                     NaN   
1                      261.0  17052889          NaN                     NaN   
2                      261.0  14186244          NaN                     NaN   
3                      261.0  12177905          NaN                     NaN   
5                      266.0  17143294          NaN                     NaN   
...                      ...       ...          ...                     ...   
71851                  261.0  12412492          NaN                     NaN   
71854                  261.0  17239384          NaN                     NaN   
71855                  261.0  12773139          NaN                     NaN   
71856                  261.0  12826780          NaN                     NaN   
71857                  261.0  13083743          NaN                     NaN   

       heatingorsystemtypeid  buildingclasstypeid  architecturalstyletypeid  \
0                        NaN                  NaN                       NaN   
1                        NaN                  NaN                       NaN   
2                        NaN                  NaN                       NaN   
3                        2.0                  NaN                       NaN   
5                        NaN                  NaN                       NaN   
...                      ...                  ...                       ...   
71851                    2.0                  NaN                       NaN   
71854                    NaN                  NaN                       NaN   
71855                    2.0                  NaN                       NaN   
71856                    2.0                  NaN                       NaN   
71857                    7.0                  NaN                       NaN   

       airconditioningtypeid       id  basementsqft  ...   id.1  logerror  \
0                        NaN  1727539           NaN  ...      0  0.025595   
1                        NaN  1387261           NaN  ...      1  0.055619   
2                        NaN    11677           NaN  ...      2  0.005383   
3                        NaN  2288172           NaN  ...      3 -0.103410   
5                        NaN  1447245           NaN  ...      5 -0.020526   
...                      ...      ...           ...  ...    ...       ...   
71851                    NaN  2274245           NaN  ...  77607  0.001082   
71854                    NaN  2968375           NaN  ...  77610  0.013209   
71855                    1.0  1843709           NaN  ...  77611  0.037129   
71856                    NaN  1187175           NaN  ...  77612  0.007204   
71857                    NaN  2455685           NaN  ...  77613 -0.197755   

       transactiondate  airconditioningdesc  architecturalstyledesc  \
0           2017-01-01                  NaN                     NaN   
1           2017-01-01                  NaN                     NaN   
2           2017-01-01                  NaN                     NaN   
3           2017-01-01                  NaN                     NaN   
5           2017-01-01                  NaN                     NaN   
...                ...                  ...                     ...   
71851       2017-09-19                  NaN                     NaN   
71854       2017-09-21                  NaN                     NaN   
71855       2017-09-21              Central                     NaN   
71856       2017-09-25                  NaN                     NaN   
71857       2018-05-25                  NaN                     NaN   

       buildingclassdesc  heatingorsystemdesc  typeconstructiondesc  \
0                    NaN                  NaN                   NaN   
1                    NaN                  NaN                   NaN   
2                    NaN                  NaN                   NaN   
3                    NaN              Central  

## Takeaways
- Before dropping nulls, we had (71858 rows, 69 cols).
- After dropping nulls, we ended with (49964 rows × 69 columns).

### Lets see what were left with

In [10]:
# a view a how many null values exist in each column
wrangle.nulls_by_columns(df).sort_values(by= 'percent', ascending=False)

count   percent
buildingclassdesc             49964  1.000000
buildingclasstypeid           49964  1.000000
finishedsquarefeet15          49964  1.000000
finishedsquarefeet13          49963  0.999980
basementsqft                  49938  0.999480
storydesc                     49938  0.999480
storytypeid                   49938  0.999480
yardbuildingsqft26            49921  0.999139
architecturalstyletypeid      49896  0.998639
architecturalstyledesc        49896  0.998639
typeconstructiontypeid        49890  0.998519
typeconstructiondesc          49890  0.998519
fireplaceflag                 49883  0.998379
finishedsquarefeet6           49824  0.997198
decktypeid                    49682  0.994356
pooltypeid10                  49626  0.993235
pooltypeid2                   49426  0.989232
poolsizesum                   49404  0.988792
hashottuborspa                49088  0.982467
yardbuildingsqft17            48246  0.965615
taxdelinquencyflag            47990  0.960492
taxdelinquencyyear            47990  0.960492
finishedsquarefeet50          45845  0.917561
finishedfloor1squarefeet      45845  0.917561
threequarterbathnbr           44463  0.889901
fireplacecnt                  43380  0.868225
pooltypeid7                   41909  0.838784
poolcnt                       41359  0.827776
airconditioningdesc           37230  0.745136
airconditioningtypeid         37230  0.745136
numberofstories               36213  0.724782
garagetotalsqft               33387  0.668221
garagecarcnt                  33387  0.668221
regionidneighborhood          31096  0.622368
buildingqualitytypeid         17053  0.341306
propertyzoningdesc            17043  0.341106
unitcnt                       17027  0.340785
heatingorsystemtypeid         16950  0.339244
heatingorsystemdesc           16950  0.339244
regionidcity                    796  0.015931
finishedsquarefeet12            141  0.002822
structuretaxvaluedollarcnt       50  0.001001
censustractandblock              36  0.000721
fullbathcnt                      12  0.000240
calculatedbathnbr                12  0.000240
regionidzip                       6  0.000120
taxamount                         3  0.000060
landtaxvaluedollarcnt             1  0.000020
taxvaluedollarcnt                 1  0.000020
transactiondate                   0  0.000000
assessmentyear                    0  0.000000
id.1                              0  0.000000
logerror                          0  0.000000
propertylandusetypeid             0  0.000000
yearbuilt                         0  0.000000
roomcnt                           0  0.000000
regionidcounty                    0  0.000000
rawcensustractandblock            0  0.000000
propertycountylandusecode         0  0.000000
parcelid                          0  0.000000
lotsizesquarefeet                 0  0.000000
longitude                         0  0.000000
latitude                          0  0.000000
fips                              0  0.000000
calculatedfinishedsquarefeet      0  0.000000
bedroomcnt                        0  0.000000
bathroomcnt                       0  0.000000
id                                0  0.000000
propertylandusedesc               0  0.000000

In [11]:
df.columns

Index(['propertylandusetypeid', 'parcelid', 'storytypeid',
       'typeconstructiontypeid', 'heatingorsystemtypeid',
       'buildingclasstypeid', 'architecturalstyletypeid',
       'airconditioningtypeid', 'id', 'basementsqft', 'bathroomcnt',
       'bedroomcnt', 'buildingqualitytypeid', 'calculatedbathnbr',
       'decktypeid', 'finishedfloor1squarefeet',
       'calculatedfinishedsquarefeet', 'finishedsquarefeet12',
       'finishedsquarefeet13', 'finishedsquarefeet15', 'finishedsquarefeet50',
       'finishedsquarefeet6', 'fips', 'fireplacecnt', 'fullbathcnt',
       'garagecarcnt', 'garagetotalsqft', 'hashottuborspa', 'latitude',
       'longitude', 'lotsizesquarefeet', 'poolcnt', 'poolsizesum',
       'pooltypeid10', 'pooltypeid2', 'pooltypeid7',
       'propertycountylandusecode', 'propertyzoningdesc',
       'rawcensustractandblock', 'regionidcity', 'regionidcounty',
       'regionidneighborhood', 'regionidzip', 'roomcnt', 'threequarterbathnbr',
       'unitcnt', 'yardbuildings

### Lets adress the rest of the  null values

### columns to remove:(column, reason for removal)
removing 
- (id, id.1, parcelid,'propertylandusetypeid','buildingqualitytypeid'), id is not necessary for our algorithms and will confuse any models from here on forward.
- (fullbathcnt,calculatedbathnbr,roomcnt),  any room room count other the bedroomcnt or bathroomcnt is not necessary considering that they return similar information if not combined info.
- (propertyzoningdesc,rawcensustractandblock,regionidcounty,censustractandblock), considering that fips is being kept for region identification purposes, these columns are not necessary.
- (assessmentyear, landtaxvaluedollarcnt, taxamount, transactiondate), considering that we have already filtered out the data to only return back information for the year 2017, and are keeping taxvaluedollarcnt, these columns are not necessary because this information can be obtained through the data that we will be keeping.
- (heatingorsystemdesc,finishedsquarefeet12,propertylandusedesc,'propertycountylandusecode','unitcnt'), calculatedfinishedsquarefeet already covers this info and heatingorsystemid already identifies this information numerically.

In [27]:
def drop_columns(df):
    df = df.drop(columns=['heatingorsystemtypeid','buildingqualitytypeid','id','parcelid','calculatedbathnbr','propertylandusetypeid','fullbathcnt','propertyzoningdesc','rawcensustractandblock','regionidcounty',
    'roomcnt','structuretaxvaluedollarcnt','assessmentyear','landtaxvaluedollarcnt','taxamount','censustractandblock',
    'id.1','transactiondate','heatingorsystemdesc','finishedsquarefeet12','propertylandusedesc','propertycountylandusecode','unitcnt'])
    return df


def split(df):
    train_and_validate, test = train_test_split(df, random_state=13, test_size=.15)
    train, validate = train_test_split(train_and_validate, random_state=13, test_size=.2)

    print('Train: %d rows, %d cols' % train.shape)
    print('Validate: %d rows, %d cols' % validate.shape)
    print('Test: %d rows, %d cols' % test.shape)
    
    return train, validate, test    

def handle_missing_values(df, prop_required_column, prop_required_row):
    #this piece of code allows us to handle the missing data and get rid of it, both in the columns and in the rows(so that we can analize better).
    print ('Before dropping nulls, %d rows, %d cols' % df.shape)
    n_required_column = round(df.shape[0] * prop_required_column)
    n_required_row = round(df.shape[1] * prop_required_row)
    df = df.dropna(axis=0, thresh=n_required_row)
    df = df.dropna(axis=1, thresh=n_required_column)
    df = drop_columns(df)
    print('After dropping nulls. %d rows. %d cols' % df.shape)
    return df

def get_exploration_data(df):
    #drops rows and columns with more than %50 data missing
    print('Before dropping nulls, %d rows, %d cols' % df.shape)
    df = handle_missing_values(df, prop_required_column=.5, prop_required_row=.5)
    print('After dropping nulls, %d rows, %d cols' % df.shape)
    
    train, validate, test = split(df)
    
    return train, validate, test
get_exploration_data(df)

Before dropping nulls, 49964 rows, 69 cols
Before dropping nulls, 49964 rows, 69 cols
After dropping nulls. 44679 rows. 12 cols
After dropping nulls, 44679 rows, 12 cols
Train: 30381 rows, 12 cols
Validate: 7596 rows, 12 cols
Test: 6702 rows, 12 cols


(       bathroomcnt  bedroomcnt  calculatedfinishedsquarefeet    fips  \
 57528          1.0         2.0                         950.0  6037.0   
 2250           3.5         5.0                        3128.0  6111.0   
 20338          2.5         3.0                        1536.0  6059.0   
 31373          2.5         2.0                        1194.0  6059.0   
 59529          2.0         4.0                        2288.0  6059.0   
 ...            ...         ...                           ...     ...   
 16134          3.0         3.0                        1581.0  6037.0   
 53368          2.0         2.0                        2258.0  6037.0   
 50836          2.0         3.0                        1992.0  6111.0   
 28844          3.0         3.0                        1915.0  6037.0   
 47863          3.0         4.0                        2570.0  6037.0   
 
          latitude    longitude  lotsizesquarefeet  regionidcity  regionidzip  \
 57528  34687977.0 -118138159.0          

# <span style="color:blue">Split Data</span>

In [30]:
# get train to expolore 
train, validate, test = get_exploration_data(df)
# seeing what the train split dataset
train.info()

Before dropping nulls, 49964 rows, 69 cols
Before dropping nulls, 49964 rows, 69 cols
After dropping nulls. 44679 rows. 12 cols
After dropping nulls, 44679 rows, 12 cols
Train: 30381 rows, 12 cols
Validate: 7596 rows, 12 cols
Test: 6702 rows, 12 cols
<class 'pandas.core.frame.DataFrame'>
Int64Index: 30381 entries, 57528 to 47863
Data columns (total 12 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   bathroomcnt                   30381 non-null  float64
 1   bedroomcnt                    30381 non-null  float64
 2   calculatedfinishedsquarefeet  30381 non-null  float64
 3   fips                          30381 non-null  float64
 4   latitude                      30381 non-null  float64
 5   longitude                     30381 non-null  float64
 6   lotsizesquarefeet             30381 non-null  float64
 7   regionidcity                  29991 non-null  float64
 8   regionidzip                   30380 non-n

In [31]:
wrangle.nulls_by_columns(train).sort_values(by= 'percent', ascending=False)

count   percent
regionidcity                    390  0.012837
regionidzip                       1  0.000033
bathroomcnt                       0  0.000000
bedroomcnt                        0  0.000000
calculatedfinishedsquarefeet      0  0.000000
fips                              0  0.000000
latitude                          0  0.000000
longitude                         0  0.000000
lotsizesquarefeet                 0  0.000000
yearbuilt                         0  0.000000
taxvaluedollarcnt                 0  0.000000
logerror                          0  0.000000

***

In [32]:
validate

bathroomcnt  bedroomcnt  calculatedfinishedsquarefeet    fips  \
50990          4.0         4.0                        2909.0  6037.0   
71369          2.0         3.0                        2201.0  6037.0   
40712          2.0         3.0                        1613.0  6111.0   
13313          3.0         5.0                        2560.0  6037.0   
15417          2.0         2.0                        1517.0  6037.0   
...            ...         ...                           ...     ...   
28605          3.0         4.0                        1779.0  6037.0   
54387          2.0         3.0                        1493.0  6037.0   
14990          2.0         2.0                        1102.0  6037.0   
34851          3.0         4.0                        2273.0  6037.0   
46623          2.0         3.0                        1392.0  6037.0   

         latitude    longitude  lotsizesquarefeet  regionidcity  regionidzip  \
50990  34154272.0 -118072688.0            11592.0       47019.0      96295.0   
71369  34068700.0 -118341707.0             7161.0       12447.0      96017.0   
40712  34205775.0 -118874393.0             9500.0       34278.0      96383.0   
13313  34071706.0 -118312205.0             9539.0       12447.0      95985.0   
15417  34188343.0 -118451323.0             7500.0       12447.0      96420.0   
...           ...          ...                ...           ...          ...   
28605  34563318.0 -118097594.0             6295.0       40227.0      97328.0   
54387  33899729.0 -118227917.0             4948.0       24174.0      96090.0   
14990  34688998.0 -118083130.0             6227.0        5534.0      97318.0   
34851  34267869.0 -118585031.0             9005.0       12447.0      96346.0   
46623  34399982.0 -118463035.0             8597.0       54311.0      96352.0   

       yearbuilt  taxvaluedollarcnt  logerror  
50990     1950.0           789122.0 -0.003082  
71369     1924.0          1305806.0  0.014879  
40712     1966.0           351004.0  0.000692  
13313     1920.0           107455.0 -0.091738  
15417     1939.0           377969.0 -0.017084  
...          ...                ...       ...  
28605     1987.0           136362.0 -0.002565  
54387     1951.0            46247.0  0.026311  
14990     1987.0           140000.0 -0.394014  
34851     1972.0           320592.0  0.012950  
46623     1966.0           196792.0 -0.070442  

[7596 rows x 12 columns]

In [33]:
test

bathroomcnt  bedroomcnt  calculatedfinishedsquarefeet    fips  \
37437          1.0         2.0                         697.0  6111.0   
51668          1.0         2.0                         924.0  6037.0   
33794          1.0         3.0                        1026.0  6037.0   
5537           2.0         3.0                        1277.0  6059.0   
65612          2.0         2.0                        1703.0  6037.0   
...            ...         ...                           ...     ...   
676            2.0         3.0                        2898.0  6037.0   
28428          3.0         3.0                        2601.0  6059.0   
57507          2.0         3.0                        1358.0  6037.0   
29286          1.0         3.0                         983.0  6037.0   
29931          2.0         4.0                        1474.0  6037.0   

         latitude    longitude  lotsizesquarefeet  regionidcity  regionidzip  \
37437  34274402.0 -119273072.0             2296.0       34543.0      97081.0   
51668  33823346.0 -118114326.0             5629.0       46298.0      96242.0   
33794  33883231.0 -118253054.0             5098.0       24174.0      96090.0   
5537   33766617.0 -117900387.0             7000.0       47568.0      97006.0   
65612  34159186.0 -118417863.0             6056.0       12447.0      96424.0   
...           ...          ...                ...           ...          ...   
676    34168443.0 -118064600.0            11770.0       47695.0      96278.0   
28428  33582006.0 -117641952.0             7000.0       12773.0      96996.0   
57507  34201617.0 -118621599.0             8447.0       12447.0      96342.0   
29286  34575829.0 -118094140.0             6089.0       40227.0      97328.0   
29931  34234603.0 -118250372.0             7469.0       45457.0      96330.0   

       yearbuilt  taxvaluedollarcnt  logerror  
37437     1928.0           327744.0  0.031970  
51668     1950.0           457702.0 -0.035976  
33794     1950.0           319803.0 -0.043604  
5537      1961.0           441633.0  0.491767  
65612     1952.0           114341.0 -0.127774  
...          ...                ...       ...  
676       1943.0           330723.0  0.028165  
28428     1994.0           497382.0 -0.057161  
57507     1956.0           100462.0  0.194552  
29286     1954.0            43506.0 -0.116010  
29931     1964.0           109099.0  0.015151  

[6702 rows x 12 columns]

***

In [ ]:
# Declaring the points for first line plot
X1 = train.bedroomcnt 
Y1 = train.fips
# plotting the first plot
plt.plot(X1, Y1, label = "plot 1") 
# Declaring the points for second line plot
X2 = train.bathroomcnt 
Y2 = train.fips
# plotting the second plot 
plt.plot(X2, Y2, label = "plot 2") 
  
# Labeling the X-axis 
plt.xlabel('X-axis') 
# Labeling the Y-axis 
plt.ylabel('Y-axis') 
# Give a title to the graph
plt.title('Two plots on the same graph') 
  
# Show a legend on the plot 
plt.legend() 
 
plt.show()

In [ ]:
train.head()
